### ISSUE TITLE
Fix `function-url-quotes` autofix for multiple `url()`

### ISSUE THREAD

**Issue by mjohenneken** [stylelint/stylelint#6710]
### What steps are needed to reproduce the bug?

Run stylint for this:

```
.foo {
        background-image: url(../img/footer/needle.svg), url(../img/footer/tel.svg);
}
```

Related standard syntax: [background-image]( https://developer.mozilla.org/en-US/docs/Web/CSS/background-image?retiredLocale=de#formal_syntax)

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard-scss"
}

```

### How did you run Stylelint?

npx stylelint template/sass/test.scss --fix

### Which version of Stylelint are you using?

`15.2.0`

### What did you expect to happen?

.foo {
        background-image: url("../img/footer/needle.svg"), url("../img/footer/tel.svg");
}

### What actually happened?

The following stylelint errors were reported:

```
template/sass/test.scss
 2:31  ✖  Expected quotes around "url" function argument  function-url-quotes
 2:62  ✖  Expected quotes around "url" function argument  function-url-quotes

2 problems (2 errors, 0 warnings)
```
With `--fix` stylelint creates malformed Sass:

```
.foo {
        background-image: url("../img/footer/needle.svg"), ur"../img/footer/tel.svg"vg);
}

```

### Does the bug relate to non-standard syntax?

no

### Proposal to fix the bug

It might have to do with multiple urls in the rule. With just one url it works.

**Comment by ybiquitous** [comment]
@mjohenneken Thanks for reporting the problem by using the template. We can reproduce the bug for `function-url-quotes` on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilAjJYA1gOYBOEArplALQCWAtksTPChaQDYAUAdLwHoAZhAi8AzgDdiASgA07Ln0H5SE6TIDcWAL5YQckELqcYAOSQM4iGAA9LAB1O8w48QfARMx4ghAZsFHQQUgpTcWC2AJwcYKEqMAAXOi8aDk4aAEcKCESYCIQg5E4AdyQATwKsHD1MHQ9IbzpiADEIUiZEvwArcS8PWAd3RGii8UTy0046TETIooBGAFZeACZeAAZguWqxiamZxJpGnxpxpGokUih54IBmDc3eBeDdEB0gA).
As you mentioned, this bug is raised only when specifying multiple `url()` with the standard CSS.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
The fix will likely involve making the autofix more robust by transforming the value AST (rather than use indexes on the declaration value string).

Depending on the primary option, we can [swap between](https://github.com/TrySound/postcss-value-parser#function) `word`  and `string` node types.

**Comment by ybiquitous** [comment]
> We'll want to make autofix more robust by transforming the value AST rather than using indexes

Agree. I'll rewrite the rule via the `postcss-value-parser` AST. 💪🏼 

But, I notice the parser cannot parse `url-prefix()` like this:

```js
const parsed = require('postcss-value-parser')('url-prefix(http://www.w3.org/Style)');
console.dir(parsed, { depth: 4 });
```

<details>
<summary>AST output</summary>

```js
ValueParser {
  nodes: [
    {
      type: 'function',
      sourceIndex: 0,
      value: 'url-prefix',
      before: '',
      after: '',
      sourceEndIndex: 35,
      nodes: [
        {
          type: 'word',
          sourceIndex: 11,
          sourceEndIndex: 15,
          value: 'http'
        },
        {
          type: 'div',
          sourceIndex: 15,
          sourceEndIndex: 16,
          value: ':',
          before: '',
          after: ''
        },
        {
          type: 'div',
          sourceIndex: 16,
          sourceEndIndex: 17,
          value: '/',
          before: '',
          after: ''
        },
        {
          type: 'div',
          sourceIndex: 17,
          sourceEndIndex: 18,
          value: '/',
          before: '',
          after: ''
        },
        {
          type: 'word',
          sourceIndex: 18,
          sourceEndIndex: 28,
          value: 'www.w3.org'
        },
        {
          type: 'div',
          sourceIndex: 28,
          sourceEndIndex: 29,
          value: '/',
          before: '',
          after: ''
        },
        {
          type: 'word',
          sourceIndex: 29,
          sourceEndIndex: 34,
          value: 'Style'
        }
      ]
    }
  ]
}
```

</details>

We have test cases about `url-prefix()`. Do you know a solution for such a case? @jeddy3 

https://github.com/stylelint/stylelint/blob/c9ad356f7df5772d549a43f33cec51ea77a37a62/lib/rules/function-url-quotes/__tests__/index.js#L201

**Comment by ybiquitous** [comment]
I'm not sure but I try creating a patch https://github.com/stylelint/stylelint/commit/4779d7fbea1941d060cc5f723420c16b62d9d3ac to fix the `url-prefix()` problem.
